aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,111 @@
+- Sets a [PostEvictionDelegate](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.postevictiondelegate) that will be called after the entry is evicted from the cache. The callback is run on a different thread from the code that removes the item from the cache.
 
-The token the item depended upon fired an event.
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_et)]
 
-**Capacity**
+## Cache dependencies
 
-The item was removed as part of the cache's memory management process.
+The following sample shows how to expire a cache entry if a dependent entry expires. A `CancellationTokenSource` is added to the cached item. When `Cancel` is called on the token, both cache entries are evicted. 
 
-You can specify that one or more cache entries depend on a `CancellationTokenSource` by adding the expiration token to the `MemoryCacheEntryOptions` object. When a cached item is invalidated, call `Cancel` on the token, which will expire all of the associated cache entries (with a reason of `TokenExpired`). The following unit test demonstrates this:
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_ed)]
 
-[!code-csharp[Main](memory/sample/test/CachingSample.Tests/MemoryCacheTests.cs?highlight=7,16,21&range=43-64)]
+Using a `CancellationTokenSource` allows multiple cache entries to be evicted as a group. With the `using` pattern in the code above, cache entries created inside the `using` block will inherit triggers and timeouts.
 
-Using a `CancellationTokenSource` allows multiple cache entries to all be expired without the need to create a dependency between cache entries themselves (in which case, you must ensure that the source cache entry exists before it is used as a dependency for other entries).
+### Addition notes
 
-Cache entries will inherit triggers and timeouts from other entries accessed while creating the new entry. This approach ensures that subordinate cache entries expire at the same time as related entries.
+- When using a callback to repopulate a cache item:
 
-[!code-csharp[Main](memory/sample/test/CachingSample.Tests/MemoryCacheTests.cs?highlight=7,11,13,23,24&range=66-94)]
+  - Multiple requests can find the cached key value empty (because the callback hasn't completed). 
+  - This can result in several threads repopulating the cached item.
 
-> [!NOTE]
-> When one cache entry is used to create another, the new one copies the existing entry's expiration tokens and time-based expiration settings, if any. It is not expired in response to manual removal or updating of the existing entry.
+- When one cache entry is used to create another, the new one copies the existing entry's expiration tokens and time-based expiration settings. The copy is not expired by manual removal or updating of the parent entry.
```

Comment:
The terms "new" and "copy" are a little confusing. I suggest using "parent" and "child" to clarify the relationship.